realm/realm-library/src/main/cpp/io_realm_internal_Collection.cpp

```diff
@@ -420,10 +350,8 @@ Java_io_realm_internal_Collection_nativeDeleteFirst(JNIEnv *env, jclass, jlong n
 
     try {
         auto wrapper = reinterpret_cast<ResultsWrapper*>(native_ptr);
-        if (wrapper->get_results().size() > 0) {
-            wrapper->get_results().get_tableview().remove(0);
-            // Refresh snapshot
-            wrapper->refresh_snapshot();
+        if (wrapper->m_results.size() > 0) {
+            wrapper->m_results.get_tableview().remove(0);
```

Comment:
Great point! And also the `get_tableview()` creates some unnecessary overhead.
Changed to use `Row::move_last_over()` instead.